poe-api/src/poe.py

```diff
@@ -221,9 +219,14 @@ def get_next_data(self, overwrite_vars=False):
     json_regex = r'<script id="__NEXT_DATA__" type="application\/json">(.+?)</script>'
     json_text = re.search(json_regex, r.text).group(1)
     next_data = json.loads(json_text)
+    key_regex = r'(https://[^<]*_app.*?\.js)'
+    key_url = re.search(key_regex, r.text).group(1)
+    s = request_with_retries(self.session.get, key_url)
+    key_regex = r'\+"([a-zA-z0-9]{17})"'
```

Comment:
This regex is way too broad. This is searching for *every* 17 character alphanumeric string, which may turn up false positives.